Source/Core/Core/NetPlayServer.cpp

```diff
@@ -2085,6 +2045,47 @@ bool NetPlayServer::PlayerHasControllerMapped(const PlayerId pid) const
          std::any_of(m_wiimote_map.begin(), m_wiimote_map.end(), mapping_matches_player_id);
 }
 
+void NetPlayServer::AssignNewUserAPad(const Client& player)
+{
+  for (PlayerId& mapping : m_pad_map)
+  {
+    if (mapping == Unmapped)
+    {
+      mapping = player.pid;
+      break;
+    }
+  }
+}
+
+PlayerId NetPlayServer::GiveFirstAvailableIDTo(ENetPeer* player)
+{
+  PlayerId pid = 1;
+  for (auto i = m_players.begin(); i != m_players.end(); ++i)
+  {
+    if (i->second.pid == pid)
+    {
+      pid++;
+      i = m_players.begin();
+    }
+  }
+  player->data = new PlayerId(pid);
+  return pid;
+}
+
+template <typename... Data>
+void NetPlayServer::SendResponse(const MessageID message_id, const Client* player,
+                                 Data&&... data_to_send)
+{
+  sf::Packet response;
+  response << message_id;
+  (response << ... << std::forward<Data>(data_to_send));
```

Comment:
Can we have a comment? something like:

`// use a c++17 fold expression to call the << operator for all the data`

Otherwise someone who doesn't know about fold expressions is going to have a really hard time trying to google what's going on here.

Reply:
Yeah, that seems reasonable. I had no clue what that was until you linked it